test/framework/clusterctl/e2e_config.go

```diff
@@ -800,6 +800,21 @@ func (c *E2EConfig) GetVariable(varName string) string {
 	return value
 }
 
+// GetVariableBestEffort returns a variable from environment variables or from the e2e config file.
+// If the variable cannot be found it returns an empty string and does not fail.
+func (c *E2EConfig) GetVariableBestEffort(varName string) string {
```

Comment:
Might be cleaner to rename
GetVariable --> MustGetVariable
GetVariableBestEffort --> GetVariable

But I'm also ok to defer this to a follow up PR

Reply:
Will do as a quick follow-up. We'll likely have to cheryr-pick this PR to be able to cherry-pick #11649 11395

Reply:
xref: https://github.com/kubernetes-sigs/cluster-api/pull/11743